qirlib/src/generation/emit.rs

```diff
@@ -67,40 +59,24 @@ fn build_entry_function(generator: &CodeGenerator, model: &SemanticModel) -> Res
     add_external_functions(generator, model.external_functions.iter());
     let entry_point = qir::create_entry_point(generator.context, &generator.module);
 
-    if generator.use_static_qubit_alloc {
-        let num_qubits = format!("{}", model.qubits.len());
+    if model.num_qubits > 0 {
         let required_qubits = generator
             .context
-            .create_string_attribute("requiredQubits", &num_qubits);
+            .create_string_attribute("requiredQubits", &model.num_qubits.to_string());
         entry_point.add_attribute(AttributeLoc::Function, required_qubits);
     }
 
-    if generator.use_static_result_alloc {
-        let num_results: u64 = model.registers.iter().map(|f| f.size).sum();
-        let num_result_string = format!("{}", num_results);
+    if model.num_results > 0 {
         let required_results = generator
             .context
-            .create_string_attribute("requiredResults", &num_result_string);
+            .create_string_attribute("requiredResults", &model.num_results.to_string());
         entry_point.add_attribute(AttributeLoc::Function, required_results);
     }
```

Comment:
I'm not sure if:
- the `requiredQubits` and `requiredResults` attributes should be omitted if the count is 0; or
- the attributes should always be present; or
- `num_qubits` and `num_results` in `SemanticModel` should be `Option<u64>`.

Reply:
Based on the other answer, I think this code can just be removed.